webclient/src/hooks/client.js

```diff
@@ -0,0 +1,71 @@
+import M from 'materialize-css';
+
+export async function client(endpoint, { body, ...customConfig } = {}) {
+  const token = localStorage.getItem('token');
+  const headers = { 'Content-Type': 'application/json' };
+
+  if (token) {
+    headers.Authorization = `${token}`;

```

Comment:
I think we should only put a token in the header if the API requires it

Reply:
if we send this everytime, then if the server returns with a 401 when the user loads homepage for the first time, we can redirect the users to re-login.

current implementation will always think that the user is logged in if token is there, while it might be that the token is stale.

Reply:
Tested during dev, any 401 server response will redirect user to relogin
- not yet done: clear token when receiving 401 response (do we want this kind of implementation?)

Reply:
Hmm, but for public pages, I think this is unnecessary
We can prompt users to login if they try to open pages with restricted access

Reply:
but it will ignore if there is no token right?